org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/ast/TypeDeclaration.java

```diff
@@ -1883,5 +1885,45 @@ public boolean isPackageInfo() {
 public boolean isSecondary() {
 	return (this.bits & ASTNode.IsSecondaryType) != 0;
 }
+public void updateSupertypesWithAnnotations(Map<ReferenceBinding,ReferenceBinding> outerUpdates) {
+	if (this.binding == null || this.binding.isHierarchyInconsistent())
+		return;
+	this.binding.getAnnotationTagBits();
+	if (this.binding instanceof MemberTypeBinding) {
+		((MemberTypeBinding) this.binding).updateDeprecationFromEnclosing();
+	}
+	Map<ReferenceBinding,ReferenceBinding> updates = new HashMap<>();
+	if (this.superclass != null) {
+		this.binding.superclass = updateWithAnnotations(this.superclass, 0, outerUpdates, updates);
+	}
+	if (this.superInterfaces != null && this.binding.superInterfaces != null && this.superInterfaces.length == this.binding.superInterfaces.length) {
+		for (int i = 0; i < this.superInterfaces.length; i++) {
+			// correspondence between arrays this.binding.superInterfaces and this.superInterfaces is ensured by !this.binding.isHierarchyInconsistent()
+			this.binding.superInterfaces[i] = updateWithAnnotations(this.superInterfaces[i], 0, outerUpdates, updates);
+		}
+	}
+	if (this.memberTypes != null) {
+		for (TypeDeclaration memberTypesDecl : this.memberTypes) {
+			memberTypesDecl.updateSupertypesWithAnnotations(updates);
+		}
+	}
+}
 
+protected ReferenceBinding updateWithAnnotations(TypeReference typeRef, int location,
+		Map<ReferenceBinding, ReferenceBinding> outerUpdates, Map<ReferenceBinding, ReferenceBinding> updates)
```

Comment:
location parameter is always 0 at this method ??

Reply:
thanks for catching this left over from previous attempts :)